lib/rules/no-unneeded-ternary.js

```diff
@@ -60,9 +78,21 @@ module.exports = {
 
             ConditionalExpression(node) {
                 if (isBooleanLiteral(node.alternate) && isBooleanLiteral(node.consequent)) {
-                    context.report(node, node.consequent.loc.start, "Unnecessary use of boolean literals in conditional expression.");
+                    context.report({
```

Comment:
Ah, good catch; for some reason I was thinking that only `foo ? true : false` would be reported by this rule.

So the cases are:

* `foo ? true : false`: do what the check currently does now
* `foo ? true : true` or `foo ? false : false`: We can just replace this with `true` or `false` respectively if `foo` is an identifier. For `foo() ? true : true` we probably shouldn't fix it.
* `foo ? false : true`: We can negate the left side and replace the node with that, although we would have to be careful about negating things like `<` operators due to `NaN`.